easybuild/easyblocks/p/parmetis.py

```diff
@@ -166,20 +168,37 @@ def install_step(self):
         # other applications depending on ParMETIS (SuiteSparse for one) look for both ParMETIS libraries
         # and header files in the Lib directory (capital L). The following symlink are hence created.
         try:
-            llibdir = os.path.join(self.installdir, 'Lib')
-            os.symlink(libdir, llibdir)
-            for f in ['metis.h', 'parmetis.h']:
-                os.symlink(os.path.join(includedir, f), os.path.join(libdir, f))
+            caplibdir = os.path.join(self.installdir, 'Lib')
+            remove_file(caplibdir)
+            symlink(libdir, caplibdir)
+            for header_file in ['metis.h', 'parmetis.h']:
+                header_path = os.path.join(libdir, header_file)
+                remove_file(header_path)
+                symlink(os.path.join(includedir, header_file), header_path)
         except OSError as err:
             raise EasyBuildError("Something went wrong during symlink creation: %s", err)
 
     def sanity_check_step(self):
         """Custom sanity check for ParMETIS."""
 
+        # Paths to libraries defined depending on SHARED configuration option
+        config_shared = False
+        config_true = ['1', 'ON', 'YES', 'TRUE', 'Y']  # True values in CMake
+        config_list = self.cfg['configopts'] if isinstance(self.cfg['configopts'], list) else [self.cfg['configopts']]
+        for configopt in config_list:
+            for opt in configopt.split():
+                if 'SHARED' in opt and any(trueval in opt for trueval in config_true):
+                    config_shared = True
+
+        if config_shared:
```

Comment:
@lexming There's one small issue here...

Even with an iterated installation (when using a list of `configopts`), the sanity check is only run once.

So, depending on whether `SHARED=1` is used first or last, you'll only check for `libparmetis.a` or `libparmetis.so`...

It would be better if you move the code that checks whether `SHARED=1` (or equivalent) was used to the `configure_step`, and set `self.config_shared` there, also introduce a `self.config_static` (and set it to `True` if `SHARED=1` was not found), both initialize them to `False` in the easyblock constructor, and then change the code here to:

```python
parmetis_libs = [os.path.join('lib', 'libmetis.a')]
if self.config_shared:
    shlib_ext = get_shared_lib_ext()
    parmetis_libs.append(os.path.join('lib', 'libparmetis.%s' % shlib_ext))
if self.config_static:
    parmetis_libs.append(os.path.join('lib', 'libparmetis.a'))

Reply:
That's very true. Moving the detection of `SHARED` to the config step also simplifies quite a bit the code, as the `configopts` will always be a string inside `config_step()`.
Fixed :+1: